### PR TITLE
[MAT-3818] Auto Logout Inactive Users

### PR DESCRIPTION
**Pull request by jkotanchik-SB**
## MADiE PR

Jira Ticket: [MAT-3818](https://jira.cms.gov/browse/MAT-3818)
(Optional) Related Tickets:

### Summary
- Move TimeoutHandler to common location and add okta logout/revoke calls
- Replace manual calls to revoke/logout with Okta React Hook
- Set inactivity timer to display warning after 25 min

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -95,6 +95,7 @@
     "react-dom": "^17.0.2",
     "single-spa": "^5.9.3",
     "single-spa-react": "^4.4.0",
+    "query-string": "^7.1.1",
```
Comment (mcmcphillips) [inline package.json]: This looks like it may no longer be needed

File: src/types/madie-timeout.d.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+declare module "@madie/madie-timeout" {
+  import { FC } from "react";
+
+  export const TimeoutHandler: FC<{ timeLeft: number }>;
+}
```
Comment (mcmcphillips) [inline src/types/madie-timeout.d.ts]: This may also no longer be needed

**Review by mcmcphillips — CHANGES_REQUESTED** [review]
This works as expected

File: package.json (hunk 1 above)
Comment (jkotanchik-SB) [reply package.json]: Agreed. Removed.

File: src/types/madie-timeout.d.ts (hunk 2 above)
Comment (jkotanchik-SB) [reply src/types/madie-timeout.d.ts]: Agreed. Removed.

File: src/custom-hooks/useOktaTokens.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+const getField = (storageKey, field) => {
```
Comment (adongare) [inline src/custom-hooks/useOktaTokens.ts]: can we not use this from madie-util?

Comment (jkotanchik-SB) [reply src/custom-hooks/useOktaTokens.ts]: Ah, this file can be deleted. It's leftover from my original attempt at autologout.

**Review by mcmcphillips — DISMISSED** [review]

**Review by adongare — DISMISSED** [review]

**Review by RohitKandimalla — APPROVED** [review]
